src/lighteval/metrics/dynamic_metrics.py

```diff
@@ -56,171 +56,178 @@
-        corpus_level_fn=np.mean,
-        higher_is_better=True,
-    )
-
-
-def multilingual_extractive_match_metric(
-    language: Language = Language.ENGLISH,
-    gold_extraction_target: Sequence[ExtractionTarget] = (ExprExtractionConfig(),),
-    pred_extraction_target: Sequence[ExtractionTarget] = (ExprExtractionConfig(), LatexExtractionConfig()),
-    aggregation_function: Callable[[list[float]], float] = max,
-    fallback_mode: Literal["no_fallback", "first_match"] = "first_match",
-    extraction_mode: Literal["first_match", "any_match"] = "any_match",
-    precision: int = 6,
-    timeout_seconds: int = 5,
-) -> SampleLevelMetric:
-    """Creates a language-aware extractive match metric that extracts answers from the model's output.
-
-    Known issues:
-    - If the task is to simplify an expression, the metric might overestimate the accuracy. This is because if the model doesn't output any anchor for the extraction (e.g final answer is..),
-        it's possible that the extracted prediction will be the expression to simplify. Because we do simplifications ourselves, it can thus happen that sympy will correctly simplify the expression,
-        thus it will match gold, despite model not doing anything. PRs to fix this are welcome.
-
-    - There is currently no StringExtractionConfig, so if the gold is \boxed{\text{Friday}} and model outputs Friday it will not match, because nothing will be extracted.
-
-    Args:
-        language: Language
-            The language of the samples.
-        gold_extraction_target: Sequence[ExtractionTarget]
-            Extraction targets to use for gold answers. Defaults to extracting simple math expressions.
-        pred_extraction_target: Sequence[ExtractionTarget]
-            Extraction targets to use for predictions. Defaults to extracting simple math expressions.
-        aggregation_function: Callable[[list[float]], float]
-            Function to aggregate scores when multiple golds/predictions are present. Defaults to max.
-        fallback_mode: Literal["no_fallback", "first_match"]
-            How to perform extraction. Defaults to "first_match".
-            - "no_fallback": Only use first successfully parsed matches
-            - "first_match": Use the first successfully parsed match + first match irregardless the parsing success
-        extraction_mode: Literal["first_match", "any_match"]
-            - "first_match": Only tries to extract the first regex match if it fails no other matches are tried
-            - "any_match": Tries to extract any regex match
-
-        precision: int
-            Number of decimal places to use when comparing numerical values. Defaults to 6.
-        timeout_seconds: int
-            Timeout for the extraction (each attempt) and comparison. Defaults to 5.
-
-    Returns:
-        A sample level metric that extracts and compares mathematical expressions.
-
-    """
+class LogLikelihoodAccMetric(SampleLevelMetric):
```

Comment:
changed all to classes to avoid nested functions which return classes - now it's clear what kind of objects they are